### PR TITLE
app-misc/anki: drop support for Qt5 GUI in favor of Qt6

### PR DESCRIPTION
**Pull request by antecrescent**
Adapt package to conform to new policy
https://wiki.gentoo.org/index.php?title=Project:Qt/Policies&diff=prev&oldid=1310293

I hope revbumping the ebuild was the correct thing to do. My rationale is that I want to propagate the change to users of `app-misc/anki[gui,-qt6]` on `emerge --update --changed-use [at]world` updates.

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

**Comment by antecrescent** [comment]
Hi @ionenwks! I created this PR after reading your email asking contributors to drop support for Qt5. Would you be interested in taking a look at this?

File: app-misc/anki/anki-24.06.3-r1.ebuild [hunk 1]
```diff
@@ -757,22 +758,10 @@ GUI_RDEPEND="
...
-		dev-qt/qtsvg:5
-		$(python_gen_cond_dep '
-			>=dev-python/PyQt5-5.15.5[gui,network,webchannel,widgets,${PYTHON_USEDEP}]
-			>=dev-python/PyQt5-sip-12.9.0[${PYTHON_USEDEP}]
-			>=dev-python/PyQtWebEngine-5.15.5[${PYTHON_USEDEP}]')
-	)
+		dev-python/waitress[${PYTHON_USEDEP}]
+		>=dev-python/PyQt6-6.6.1[gui,network,opengl,quick,webchannel,widgets,${PYTHON_USEDEP}]
+		>=dev-python/PyQt6-sip-13.6.0[${PYTHON_USEDEP}]
+		>=dev-python/PyQt6-WebEngine-6.6.0[widgets,${PYTHON_USEDEP}]')
```
Comment (a17r) [inline app-misc/anki/anki-24.06.3-r1.ebuild:764]:
```suggestion
		>=dev-python/PyQt6-WebEngine-6.6.0[widgets,${PYTHON_USEDEP}]
	')
```

Comment (a17r) [inline app-misc/anki/anki-24.06.3-r1.ebuild:764]: that block could use some better sorting after the merging from Qt separation into the main block.

File: app-misc/anki/anki-24.06.3-r1.ebuild [hunk 2]
```diff
@@ -797,9 +786,9 @@ BDEPEND="
 		app-arch/unzip
 		>=net-libs/nodejs-20.12.1
 		sys-apps/yarn
-		$(python_gen_cond_dep 'dev-python/wheel[${PYTHON_USEDEP}]')
-		qt6?	( $(python_gen_cond_dep 'dev-python/PyQt6[${PYTHON_USEDEP}]') )
-		!qt6?	( $(python_gen_cond_dep 'dev-python/PyQt5[${PYTHON_USEDEP}]') )
+		$(python_gen_cond_dep '
+		dev-python/wheel[${PYTHON_USEDEP}]
+		dev-python/PyQt6[${PYTHON_USEDEP}]')
```
Comment (a17r) [inline app-misc/anki/anki-24.06.3-r1.ebuild:791]:
```suggestion
		$(python_gen_cond_dep '
			dev-python/wheel[${PYTHON_USEDEP}]
			dev-python/PyQt6[${PYTHON_USEDEP}]
		')
```

File: app-misc/anki/anki-24.06.3-r1.ebuild [hunk 3]
```diff
@@ -964,16 +947,16 @@ src_install() {
...
 	if use gui; then
 		xdg_pkg_postinst
 		optfeature "LaTeX in cards" "app-text/texlive[extra] app-text/dvipng"
 		optfeature "sound support" media-video/mpv media-video/mplayer
-		optfeature "recording support" "media-sound/lame[frontend] dev-python/PyQt$(usex qt6 6 5)[multimedia]"
+		optfeature "recording support" "media-sound/lame[frontend] dev-python/PyQt6[multimedia]"
 		optfeature "faster database operations" dev-python/orjson
-		use qt6 && optfeature "compatibility with Qt5-dependent add-ons" dev-python/PyQt6[dbus,printsupport]
-		use qt6 && optfeature "Vulkan driver" "media-libs/vulkan-loader dev-qt/qtbase[vulkan]
+		optfeature "compatibility with Qt5-dependent add-ons" dev-python/PyQt6[dbus,printsupport]
```
Comment (a17r) [inline app-misc/anki/anki-24.06.3-r1.ebuild:958]:
```suggestion
		optfeature "compatibility with Qt6-dependent add-ons" dev-python/PyQt6[dbus,printsupport]
```

**Review by a17r — CHANGES_REQUESTED** [review]
> Hi @ionenwks! I created this PR after reading your email asking contributors to drop support for Qt5. Would you be interested in taking a look at this?

That doesn't mean they shall do all the ensuing reviews ;)

File: app-misc/anki/anki-24.06.3-r1.ebuild (hunk 3 above)
Comment (antecrescent) [reply app-misc/anki/anki-24.06.3-r1.ebuild:962]: `Qt5-dependent add-ons` is actually correct. If `ENABLE_QT5_COMPAT` is set at runtime Anki imports the [`qt5_compat`](https://github.com/ankitects/anki/blob/24.06.3/qt/aqt/qt/qt5_compat.py) module, which pulls in the stated runtime dependency.
Said module exports the PyQt5 API by creating aliases to PyQt6 among other things. This way, older Anki add-ons that rely on the PyQt5 API keep working with the Qt6 GUI.

**Comment by antecrescent** [comment]
> > Hi @ionenwks! I created this PR after reading your email asking contributors to drop support for Qt5. Would you be interested in taking a look at this?
> 
> That doesn't mean they shall do all the ensuing reviews ;)

You are right, sorry.

**Comment by antecrescent** [comment]
@a17r I implemented your requested changes. Is there anything else I can do for you?

**Review by juippis — APPROVED** [review]

**Pull request closed without merging**